packages/node/e2e/authenticated.test.ts

```diff
@@ -0,0 +1,201 @@
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { join } from "path";
+import {
+  test,
+  pass,
+  beforeEach,
+  afterEach,
+  strictSame,
+  strictNotSame,
+  resolveMatch,
+} from "tap";
+
+import { config } from "dotenv-flow";
+import { custom } from "openid-client";
+import { Session } from "../src/Session";
+
+custom.setHttpOptionsDefaults({
+  timeout: 15000,
+});
+
+// Load environment variables from .env.local if available:
+config({
+  path: join(__dirname, "..", "..", "e2e"),
+  // In CI, actual environment variables will overwrite values from .env files.
+  // We don't need warning messages in the logs for that:
+  silent: process.env.CI === "true",
+});
+
+type AuthDetails = {
+  pod: string;
+  oidcIssuer: string;
+  clientId: string;
+  clientSecret: string;
+};
+// Instructions for obtaining these credentials can be found here:
+// https://github.com/inrupt/solid-client-authn-js/blob/1a97ef79057941d8ac4dc328fff18333eaaeb5d1/packages/node/example/bootstrappedApp/README.md
+const serversUnderTest: AuthDetails[] = [
+  // pod.inrupt.com:
+  {
+    // Cumbersome workaround, but:
+    // Trim `https://` from the start of these URLs,
+    // so that GitHub Actions doesn't replace them with *** in the logs.
+    pod: process.env.E2E_TEST_ESS_POD!.replace(/^https:\/\//, ""),
+    oidcIssuer: process.env.E2E_TEST_ESS_IDP_URL!.replace(/^https:\/\//, ""),
+    clientId: process.env.E2E_TEST_ESS_CLIENT_ID!,
+    clientSecret: process.env.E2E_TEST_ESS_CLIENT_SECRET!,
+  },
+  // dev-next.inrupt.com:
+  {
+    // Cumbersome workaround, but:
+    // Trim `https://` from the start of these URLs,
+    // so that GitHub Actions doesn't replace them with *** in the logs.
+    pod: process.env.E2E_TEST_DEV_NEXT_POD!.replace(/^https:\/\//, ""),
+    oidcIssuer: process.env.E2E_TEST_DEV_NEXT_IDP_URL!.replace(
+      /^https:\/\//,
+      ""
+    ),
+    clientId: process.env.E2E_TEST_DEV_NEXT_CLIENT_ID!,
+    clientSecret: process.env.E2E_TEST_DEV_NEXT_CLIENT_SECRET!,
+  },
+  // inrupt.net
+  // Unfortunately we cannot authenticate against Node Solid Server yet, due to this issue:
+  // https://github.com/solid/node-solid-server/issues/1533
+  // Once that is fixed, credentials can be added here, and the other `describe()` can be removed.
+];
+
+serversUnderTest.forEach((envValues) => {
+  const {
+    clientId,
+    clientSecret,
+    oidcIssuer: oidcIssuerDisplay,
+    pod: podDisplay,
+  } = envValues;
+  console.log(
+    `Running tests for ${podDisplay}, authenticated to ${oidcIssuerDisplay}`
+  );
+  const oidcIssuer = `https://${oidcIssuerDisplay}`;
+  const podRoot = `https://${podDisplay}`;
+
+  const session = new Session();
+
+  beforeEach(async () => {
+    await session.login({
+      clientId,
+      clientSecret,
+      oidcIssuer,
+    });
+  });
+
+  afterEach(async () => {
+    await session.logout();
+  });
+
+  test("properly sets up session information", async () => {
+    strictSame(
```

Comment:
There is also a mocha like DSL, which might make these easier? https://node-tap.org/docs/api/mochalike/

Reply:
I don't really have an opinion on which one reads better, as in any case it's different from the `expect(...).toStrictEqual(...)` from Jest. I'm happy to switch to the Mocha-like DSL if you think it improves readability.

Reply:
I think it might be good, just so there's readable consistency between these tests and our other tests — less to learn, less context switching. Thoughts @matthieubosquet & @ajacksified ?

Reply:
I personally prefer the mocha-like syntax, although it's going to be a context switch either way. I could go in either direction.

Reply:
I prefer consistency and the mocha like syntax.

Reply:
That's settled then !